evalml/tests/data_checks_tests/test_target_leakage_data_check.py

```diff
@@ -621,3 +621,12 @@ def test_target_leakage_none_pearson():
     expected = {"warnings": [], "errors": [], "actions": []}
 
     assert leakage_check.validate(X, y) == expected
+
+
+def test_target_leakage_maintains_logical_types():
+    X = pd.DataFrame({"A": pd.Series([1, 2, 3]), "B": pd.Series([4, 5, 6])})
+    y = pd.Series([1, 2, 3])
+
+    X.ww.init(logical_types={"A": "Unknown", "B": "Double"})
+    warnings = TargetLeakageDataCheck().validate(X, y)["warnings"]
+    assert not any(w["message"].startswith("Column 'A'") for w in warnings)
```

Comment:
Maybe we should add a comment explaining that mutual information is not supported for Unknown logical types so they should not be included.

I'm also thinking we should just mock `ww.mutual information()` and verify the logical types are consistent there? Not sure how tricky that would be though.